src/groovy/org/pih/warehouse/api/StockMovement.groovy

```diff
@@ -234,9 +234,8 @@ class StockMovement {
 
     Boolean isReceivingAuthorized(Location currentLocation) {
         boolean isSameDestination = destination?.id == currentLocation?.id
-        boolean hasBeenPlaced = hasBeenShipped() || hasBeenPartiallyReceived()
 
-        return !hasBeenReceived() && hasBeenPlaced && (hasBeenIssued() || hasBeenShipped()) && isSameDestination
+        return !hasBeenReceived() && (hasBeenIssued() || hasBeenShipped() || hasBeenPartiallyReceived()) && isSameDestination
```

Comment:
I feel like this is getting out of hand a bit. I think there should be at least few unit tests for checking if receiving is authorized basing on Katarzyna's test cases because I feel like this is going back and forth too much.

Reply:
Yeah, you're probably right. There are too many cases for both receiving and editing.

Reply:
Agreed. @DominikaHatala Can you add a tech debt ticket for adding unit tests?